doc/Command-Reference.md

```diff
@@ -7961,8 +7962,292 @@ Go Back To [Beginning of the document](#) or [Beginning of this section](#waterm
+
+    --help                          Show this message and exit.
+  ```
+- Example:
+  ```
+  admin@sonic:~$ sudo sonic-package-manager install dhcp-relay==1.0.2
+  ```
+  ```
+  admin@sonic:~$ sudo sonic-package-manager install --from-repository azure/sonic-cpu-report:latest
+  ```
+  ```
+  admin@sonic:~$ sudo sonic-package-manager install --from-tarball sonic-docker-image.gz
+  ```
+
+**sonic-package-manager uninstall**
+
+This command uninstalls package from SONiC host. *NOTE*: this command requires elevated (root) privileges to run.
+
+- Usage:
+  ```
+  Usage: sonic-package-manager uninstall [OPTIONS] NAME
+
+    Uninstall package
+
+  Options:
+    -f, --force          Force operation by ignoring failures
+    -y, --yes            Automatically answer yes on prompts
+    -v, --verbosity LVL  Either CRITICAL, ERROR, WARNING, INFO or DEBUG
+    --help               Show this message and exit.
+
+  ```
+- Example:
+  ```
+  admin@sonic:~$ sudo sonic-package-manager uninstall dhcp-relay
+  ```
+
+**sonic-package-manager upgrade**
+
+This command upgrades package on SONiC host to a newer version. The procedure of upgrading a package will restart the corresponding service. *NOTE*: this command requires elevated (root) privileges to run.
+
+- Usage:
+  ```
+  Usage: sonic-package-manager upgrade [OPTIONS] [PACKAGE_EXPR]
+
+    Upgrade package
+
+  Options:
+    --from-repository TEXT          Fetch package directly from image registry
+                                    repository NOTE: This argument is mutually
+                                    exclusive with arguments: [package_expr,
+                                    from_tarball].
+
+    --from-tarball FILE             Fetch package from saved image tarball
+                                    NOTE: This argument is mutually exclusive
+                                    with arguments: [package_expr,
+                                    from_repository].
+
+    -f, --force                     Force operation by ignoring failures
+    -y, --yes                       Automatically answer yes on prompts
+    -v, --verbosity LVL             Either CRITICAL, ERROR, WARNING, INFO or
+                                    DEBUG
+
+    --skip-cli-plugin-installation  Do not install CLI plugins provided by the
+                                    package on the host OS. NOTE: In case when
+                                    package /cli/mandatory field is set to True
+                                    this option will fail the installation.
+
+    --help                          Show this message and exit.
+  ```
+- Example:
+  ```
+  admin@sonic:~$ sudo sonic-package-manager upgrade dhcp-relay==2.0.0
+  ```
+  ```
+  admin@sonic:~$ sudo sonic-package-manager upgrade --from-repository azure/sonic-cpu-report:latest
+  ```
+  ```
+  admin@sonic:~$ sudo sonic-package-manager upgrade --from-tarball sonic-docker-image.gz
+  ```
+
+**sonic-package-manager show package versions**
+
```

Comment:
Is this the list of all images locally available in the device ?

If yes, does this mean, each time a image is upgraded, the older one is retained ?
else which repo being referred?